cgroups/fs/apply_raw.go

```diff
@@ -27,18 +27,20 @@ var (
 // The absolute path to the root of the cgroup hierarchies.
 var cgroupRoot string
 
-// TODO(vmarmol): Report error here, we'll probably need to wait for the new API.
-func init() {
+// Sets the cgroupRoot.
+func setCgroupRoot() error {
 	// we can pick any subsystem to find the root
 	cpuRoot, err := cgroups.FindCgroupMountpoint("cpu")
 	if err != nil {
-		return
+		return err
 	}
 	cgroupRoot = filepath.Dir(cpuRoot)
 
 	if _, err := os.Stat(cgroupRoot); err != nil {
-		return
+		return err
```

Comment:
Don't know how it was before, but if `cgroupRoot` is not `""` and `os.Stat` failed, then second time `setCgroupRoot` won't be called. Also this function now can be called from concurrent goroutines.
So, resume:
* we need set cgroupRoot back to `""` on `Stat` fail
* we need to protect it with mutex